src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -93,6 +97,7 @@ public string Path
                 throw new ArgumentNullException(nameof(context));
             }
 
+            _bufferSource = context.HttpContext.RequestServices.GetRequiredService<IRazorBufferSource>();
```

Comment:
- Thread A grabs its buffer source and sets `_bufferSource`.
- Thread B grabs its buffer source and sets `_bufferSource`.
- Thread A gets around to an invocation of `RenderPageAsync()`, perhaps for a layout.